llpc/context/llpcCompiler.cpp

```diff
@@ -728,6 +729,218 @@ template <typename T> static FormattedBytes formatBytesLittleEndian(ArrayRef<T>
   return format_bytes(bytes, /* FirstByteOffset = */ None, /* NumPerLine = */ 16, /* ByteGroupSize = */ 1);
 }
 
+// =====================================================================================================================
+// Helper function for dumping compiler options
+//
+// @param pipelineDumpFile : Handle of pipeline dump file
+void Compiler::dumpCompilerOptions(void *pipelineDumpFile) {
+  if (!pipelineDumpFile)
+    return;
+  std::string extraInfo;
+  raw_string_ostream os(extraInfo);
+  os << ";Compiler Options: " << join(m_options, " ");
+  os.flush();
+  PipelineDumper::DumpPipelineExtraInfo(reinterpret_cast<PipelineDumpFile *>(pipelineDumpFile), &extraInfo);
+}
+
+// =====================================================================================================================
+// Build unlinked shader to ElfPackage with pipeline info.
+//
+// @param pipelineInfo : Info to build this shader module
+// @param [out] pipelineOut : Output of building this shader module
+// @param stage : Shader stage of needing to compile
+// @param pipelineDumpFile : Handle of pipeline dump file; It is optional, the default value is nullptr
+// @returns : Result::Success if successful. Other return codes indicate failure.
+Result Compiler::buildGraphicsShaderStage(const GraphicsPipelineBuildInfo *pipelineInfo,
+                                          GraphicsPipelineBuildOut *pipelineOut, UnlinkedShaderStage stage,
+                                          void *pipelineDumpFile) {
+  if (!pipelineInfo->pfnOutputAlloc)
+    return Result::ErrorInvalidPointer;
+
+  // clang-format off
+  SmallVector<const PipelineShaderInfo *, ShaderStageGfxCount> shaderInfo = {
+    &pipelineInfo->vs,
+    &pipelineInfo->tcs,
+    &pipelineInfo->tes,
+    &pipelineInfo->gs,
+    &pipelineInfo->fs,
```

Comment:
Are you sure you want every shader stage from the pipeline info?  What if all you are doing is the vertex processing stage?

Reply:
I see where the stages get cleared out now.  Sorry.  This is fine.